schemas/stsci.edu/asdf/0.1.0/core/asdf.yaml

```diff
@@ -11,6 +11,20 @@ description: |
 tag: "tag:stsci.edu:asdf/0.1.0/core/asdf"
 type: object
 properties:
+  asdf_library:
+    description: |
+      Describes the ASDF library that produced the file.
+    $ref: software
+
+  software:
+    description: |
+      Lists any relevant software that produced or modified the ASDF
+      file.  May include such things as data calibration pipelines,
+      data analysis tools etc.
+    type: array
+    items:
+      $ref: "software"
```

Comment:
Or to put another way, as defined this is a little ambiguous: Should it be a set, excluding uniques, of all the software that touched the file?  Does the order of the entries mean anything?  

I think by making this into the official history log of the file the intent is clearer, in that it will list each software any number of times each time it updated the file.  Having this baked into the file format could be a big win for provenance too.